### ISSUE TITLE
Stop gradients

### ISSUE THREAD

**Issue by ikostrikov** [twitter/torch-autograd#144]
Thanks for this awesome library!

I wonder if it is possible to stop gradients in a subgraph? I'm looking for something similar to https://www.tensorflow.org/versions/r0.9/api_docs/python/train.html#stop_gradient

**Comment by alexbw** [comment]
This is definitely possible! Is this the kind of thing that you're looking for?

```lua
autograd = require 'autograd'

-- Build a little dummy module, with an identity function
myfns = {}
myfns.block = function(x)
   return x
end

-- Now, futz with the gradients of our dummy module
autograd.overload.module("myfns", myfns, function(module)
   module.gradient("block", {
      function(g, ans, x) return 0 end,
   })
end)

-- Build a function we would like the derivative of, with a switch
-- to easily turn on and off our blocking function
function f(x,doblock)
   if doblock then
      return myfns.block(torch.sum(x))
   else
      return torch.sum(x)
   end
end

-- Compare the results
g = autograd(f)
print(g(torch.eye(3),true))
print(g(torch.eye(3),false))
```

Returns:
```
-- With block on 
 0  0  0
 0  0  0
 0  0  0
[torch.DoubleTensor of size 3x3]

3
-- With block off
 1  1  1
 1  1  1
 1  1  1
[torch.DoubleTensor of size 3x3]

3
```

**Comment by alexbw** [comment]
And note that the `myfns.block` function can contain anything, really. This is the mechanism you would use to define your own gradient for a block of Torch functions or to implement REINFORCE, or to debug gradient magnitude/variance of a specific partial derivative, or relax a non-differentiable function with a continuous approximation (e.g. backpropagating through `round` by either using re-centering and softmax'ing, or giving it the gradient of the identity function)

**Comment by ikostrikov** [comment]
Thanks a lot! That's exactly what a was looking for.
